### PR TITLE
grpc-thrift compat

### PR DESCRIPTION
**Pull request by chedeti**
For Code review

### CODE REVIEW

**Comment by chedeti** [comment]
Link to thrift codegen https://github.com/chedeti/thrift/pull/1

File: .gitmodules [hunk 1]
```diff
@@ -17,3 +17,6 @@
 [submodule "third_party/nanopb"]
 	path = third_party/nanopb
 	url = https://github.com/nanopb/nanopb.git
+[submodule "tools/thrift"]
```
Comment (vjpai) [inline .gitmodules]: I think we keep all external modules in third_party/ . Any reason to put this in tools/ ?

File: .gitmodules [hunk 2]
```diff
@@ -17,3 +17,6 @@
...
+	path = tools/thrift
+	url = https://github.com/chedeti/thrift.git
```
Comment (vjpai) [inline .gitmodules]: Why is the module specified like this? If it's your thrift code generation system, then that belongs as actual contents of the repository in tools/thrift , not as a submodule.

Comment (vjpai) [inline .gitmodules]: In general, we only use submodules to represent actual code taken from other open-source systems, not from grpc contributors intending to support grpc uses.

File: build.yaml [hunk 3]
```diff
@@ -1184,6 +1184,13 @@ libs:
   - grpc_test_util
   - grpc++_test_util
   - grpc++
+- name: thrift_util
+  build: all
+  language: c++
+  public_headers:
+  - include/grpc++/impl/codegen/thrift_serializer.h
+  - include/grpc++/impl/codegen/thrift_serializer_inl.h
+  - include/grpc++/impl/codegen/thrift_utils.h
```
Comment (vjpai) [inline build.yaml]: As you've already seen, this doesn't make sense as a headers-only library. I understand that your purpose is to get it included as part of the installation. You should follow the same steps used for proto_utils.h ; define the name, don't put a build: all on it, and make sure that it gets added to some other dependence chain.

**Comment by vjpai** [comment]
I really don't see the point of 57cc8f8 . Why isn't this a submodule? I don't think we should carry it around like this - better to make it a submodule pinned at a specific commit and then we can only take it when we want to. I wasn't opposed to it being a submodule - I was opposed to chedeti/thrift.git being the submodule. Let it be a submodule of the actual thrift repo pinned at a specific commit and then just check in your changes. Bringing their entire repo into ours is going to make maintainability impossible.

**Comment by pgrosu** [comment]
Dude, seriously!  Who do you expect to properly review 2,104 files?!?!?  Yes, I know it's funny to create this set of commits, but still...  

This PR definitely needs to be broken up into smaller, more manageable chunks.  Even GitHub is complaining:

github-too-many-files

**Comment by chedeti** [comment]
I am sorry, give me sometime. I am working on it.

**Comment by pgrosu** [comment]
@chedeti No worries, it's understandable :)  Thank you for helping us out.

**Pull request closed without merging**